src/chunk-stream.iced

```diff
@@ -20,28 +28,42 @@ exports.ChunkStream = class ChunkStream extends stream.Transform
     # if we don't have a full block, dump everything into extra and skip this round
     if chunk.length < @block_size
       @extra = chunk
-    else
-      # if we have to, call the transform function multiple times until we have less than a block size remaining
-      if @exact_chunking
-        while chunk.length >= @block_size
-          @push(@transform_func(chunk[...@block_size]))
-          chunk = chunk[@block_size...]
-        @extra = chunk
-      # if the transform function can accept chunks of length block_size*n for n \in N, just make one transform function call
-      else
-        remainder = chunk.length % @block_size
-        # chop off extra
-        if remainder isnt 0
-          @extra = chunk[chunk.length-remainder...]
-          chunk = chunk[...chunk.length-remainder]
-        @push(@transform_func(chunk))
-
-    cb()
+      return cb(null, new Buffer(''))
+
+    # next, mangle the chunk so that it is an even multiple of the block size
```

Comment:
What, lines 33-36? Those are still necessary, otherwise we'd be blindly calling `_transform_func` on an incomplete chunk at the end (which we should be saving).